public/expressions/overlay_anomalies.ts

```diff
@@ -77,8 +72,18 @@ const getAnomalies = async (
   return parsePureAnomalies(anomalySummaryResponse);
 };
 
+const getDetectorName = async (detectorId: string) => {
+  const resp = await getClient().get(`..${AD_NODE_API.DETECTOR}/${detectorId}`);
```

Comment:
Just to confirm, if there is failures coming from the cluster will this throw an error, or just store the error in `resp`? I Imagine we want to make sure an error is thrown so its caught in the `catch` block you have to process and produce an error `VisLayer`

Reply:
good catch, I actually fixed this by moving the call to getDetectorName inside the try, catch. Now if there is an error from the cluster it is automatically caught in the catch (like  status 4xx). If we get some response but no detector name, I manually throw the error which gets translated to a fetching failure.

Reply:
It still seems a little hacky to be basing success/failure on if the name is empty or not. In fact, can we just use existing redux functions here to handle the errors for us? For example, [getDetector](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/main/public/redux/reducers/ad.ts#L391)? We can then process any errors seen, similar to [this example](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/main/public/pages/ReviewAndCreate/containers/ReviewAndCreate.tsx#L184-L199) with `then()`/`catch()` blocks. Lmk what you think.

Reply:
Similar thought regarding the `anomalySummaryResponse` on L65. Maybe we can use the [searchResults](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/main/public/redux/reducers/anomalyResults.ts#L124) redux fn.

Reply:
I think the original intention of using `client()` standalone here was if other cluster calls needed to happen. But given that we only need standard ones we already have that are integrated with redux's already-consumed HTTP client, I think we're ok.

Reply:
I have tried using the dispatch here in a few several ways and I keep coming back to the fact that we need to wrap the class in a provider in order to use our redux methods. The error I get here is: `Uncaught Error: could not find react-redux context value; please ensure the component is wrapped in a <Provider>`. All our other uses of redux is within the parent component `<Main>` which is wrapped by the redux provider.

Reply:
Do you see other ways we could optimize this to make you more confident in just using the client? I believe the current catching mechanism works as if there is an actual error from the client like a network issue it gets caught in the catch and I have tested this (with mistyping the API for example which gave an error) and if there is some other reason the name can't be returned (I doubt this will actually happen) then that also gets caught.

Reply:
Great - I think that's fine to just use the client here. Given that this is a unique context not wrapped in the provider, I think that's ok.

As long as we have sufficient error catching & handling I'm ok with this